eureka-client/src/main/java/com/netflix/discovery/shared/resolver/aws/EurekaHttpResolver.java

```diff
@@ -0,0 +1,121 @@
+package com.netflix.discovery.shared.resolver.aws;
+
+import com.netflix.appinfo.AbstractEurekaIdentity;
+import com.netflix.appinfo.InstanceInfo;
+import com.netflix.discovery.EurekaClientConfig;
+import com.netflix.discovery.shared.Application;
+import com.netflix.discovery.shared.resolver.ClusterResolver;
+import com.netflix.discovery.shared.resolver.EurekaEndpoint;
+import com.netflix.discovery.shared.resolver.ResolverUtils;
+import com.netflix.discovery.shared.transport.EurekaHttpClientFactory;
+import com.netflix.discovery.shared.transport.EurekaHttpResponse;
+import com.netflix.discovery.shared.transport.decorator.EurekaHttpClientDecorator;
+import com.netflix.discovery.shared.transport.decorator.RetryableEurekaHttpClient;
+import com.netflix.discovery.shared.transport.decorator.ServerStatusEvaluator;
+import com.netflix.discovery.shared.transport.jersey.JerseyEurekaHttpClientFactory;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.annotation.Nullable;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+
+/**
+ * @author David Liu
+ */
+public class EurekaHttpResolver implements ClusterResolver<AwsEndpoint> {
+    private static final Logger logger = LoggerFactory.getLogger(EurekaHttpResolver.class);
+
+    private final EurekaClientConfig clientConfig;
+    private final InstanceInfo myInstanceInfo;
+    private final String appName;
+    private final EurekaHttpClientFactory clientFactory;
+
+    public EurekaHttpResolver(EurekaClientConfig clientConfig,
+                              InstanceInfo myInstanceInfo,
+                              ClusterResolver<EurekaEndpoint> bootstrapResolver,
+                              String appName) {
+        this.clientConfig = clientConfig;
+        this.myInstanceInfo = myInstanceInfo;
+        this.appName = appName;
+        this.clientFactory = RetryableEurekaHttpClient.createFactory(
+                bootstrapResolver,
+                JerseyEurekaHttpClientFactory.create(clientConfig, myInstanceInfo, resolverIdentity),
```

Comment:
This factory creates Jersey connection pool, and must be shutdown. We could create a factory in getClusterEndpoints, and shut id down immediately after usage.